### ISSUE TITLE
Lock doesn't appear to be released after failed restore

### ISSUE THREAD

**Issue by mholt** [restic/restic#708]
# Output of `restic version`
restic 0.3.2 (v0.3.2-4-gb0997d0)
compiled with go1.7.4 on darwin/amd64

## Expected behavior
An exclusive lock should not be held after an operation is complete and the process terminates.

## Actual behavior

```
$ restic restore latest --path "path/that's/not/found" --target ~/restoretest/
latest snapshot for criteria not found: no snapshot found Paths:[path/that's/not/found] Host:

$ restic check
Create exclusive lock for repository
repository is already locked by PID 57750 on Shadowfax.home by matt (UID 501, GID 20)
lock was created at 2016-12-27 08:33:22 (50.993236973s ago)
storage ID 082d53a5
the `unlock` command can be used to remove stale locks
```

## Steps to reproduce the behavior

As far as I know, just try to do a restore on a path that's not found, and then run `restic check` and it seems the lock didn't release after the failed restore. The process with ID 57750 had terminated nearly a minute earlier.

The `unlock` command did the trick but it seems a bit unnecessary? Should I just run `restic unlock` after every restic command or is that a bad idea. (Going for unsupervised execution here.)

**Comment by fd0** [comment]
Ha, thanks for the report. This is indeed a bug. Apart from when a panic occurs, restic should clean up locks. Please note that the `restore` command created a non-exclusive lock, which was left in the repo. The `check` command (by default) creates an exclusive lock.

**Comment by yhafri** [comment]
Is this bug fixed?

**Comment by fd0** [comment]
Yes, it was resolved in #709.
